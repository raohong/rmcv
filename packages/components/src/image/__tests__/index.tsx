import { render, screen } from '@testing-library/react';
import { Image, imageClassNames } from '..';

const testId = 'image';
const height = 100;

it('render correctly', () => {
  const tree = render(<Image height={height} />);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with fit', () => {
  render(<Image height={height} fit='contain' data-testid={testId} />);

  expect(screen.getByTestId(testId).querySelector('img')).toHaveStyleRule(
    'object-fit',
    'contain',
  );
});

it('render with position', () => {
  render(<Image height={height} position='left' data-testid={testId} />);

  expect(screen.getByTestId(testId).querySelector('img')).toHaveStyleRule(
    'object-position',
    'left',
  );
});

it('render with radius', () => {
  render(<Image height={height} radius={10} data-testid={testId} />);

  expect(screen.getByTestId(testId)).toHaveStyleRule('border-radius', '10px');
});

it('render with round', () => {
  render(<Image height={height} round data-testid={testId} />);

  expect(screen.getByTestId(testId)).toHaveStyleRule('border-radius', '50%');
});

it('render with showLoading', () => {
  render(<Image height={height} showLoading={false} round data-testid={testId} />);

  expect(
    screen.getByTestId(testId).querySelector(`.${imageClassNames.placeholder}`),
  ).toBeNull();
});
