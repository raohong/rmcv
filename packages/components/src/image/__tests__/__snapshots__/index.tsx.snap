// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
  display: inline-block;
  min-width: 1px;
  min-height: 1px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-1 {
  box-sizing: border-box;
  display: block;
  width: 100%;
  height: 100%;
}

<div
    class="RVuiImage-root RVuiImage-statusNone emotion-0"
    style="width: 100%; height: 100px;"
  >
    <img
      alt=""
      class="RVuiImage-img emotion-1"
      draggable="false"
    />
  </div>
</DocumentFragment>
`;
